src/library/services/httpServices/http-data-provider-service.ts

```diff
@@ -0,0 +1,70 @@
+import {Injectable} from '@angular/core';
+import {Book} from "../../model/book";
+import {DataProvider, SearchCriteria} from "../data-provider.service";
+import {HttpClient} from "@angular/common/http";
+import {map} from "rxjs/operators";
+import {Observable} from "rxjs";
+import {environment} from "../../../environments/environment";
+
+type BookPredicate = (book: Book) => boolean;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpDataProvider extends DataProvider {
+  private books: Array<Book>;
+
+  constructor(private httpClient: HttpClient) {
+    super();
+  }
+
+  public findBooks(searchCriteria: SearchCriteria): Observable<Array<Book>> {
+    let body = searchCriteria || {};
+    if (searchCriteria) {
+      return this.httpClient.post<Array<Object>>(`${environment.URL}/books/search`, body)
```

Comment:
missing required Accept header. http client implicitly use json media type, but it's recommended to specify explicitly what media type client is expecting from server